code/modules/admin/species_ban.dm

```diff
@@ -0,0 +1,107 @@
+// Species ban system.
+// Disallows a player from using a specific species, but lets them play the game regardless
+// Useful if someone isnt playing a species properly at all (IE: Not following hard-driven species lore)
+
+/// Global list of all species bans. Key is a ckey, value is a list of species. Its a global list to avoid mass DB lookups.
+GLOBAL_LIST_EMPTY(species_bans_assoc)
+//GLOBAL_PROTECT(species_bans_assoc) // Mainly just to save space in the already bloated VV window
+
+/datum/admins/proc/add_species_ban(ckey, species)
+	if(!ckey || !species)
+		return
+	if(!GLOB.species_bans_assoc["[ckey]"]) // Incase someone has a ckey thats just all numbers
+		GLOB.species_bans_assoc["[ckey]"] = list()
+
+	GLOB.species_bans_assoc["[ckey]"] |= species // Only add once
+
+/datum/admins/proc/remove_species_ban(ckey, species)
+	if(!ckey || !species)
+		return
+	if(!GLOB.species_bans_assoc["[ckey]"]) // Incase someone has a ckey thats just all numbers
+		GLOB.species_bans_assoc["[ckey]"] = list()
+
+	GLOB.species_bans_assoc["[ckey]"] -= species
+
+// Global so it can be called from world/New()
+/proc/load_all_species_bans()
+	var/datum/db_query/select_query = SSdbcore.NewQuery("SELECT ckey, job FROM [format_table_name("ban")] WHERE bantype='SPECIES_BAN' AND isnull(unbanned)")
+
+	if(!select_query.warn_execute(async = FALSE)) // NOT async because its called in world/New()
+		qdel(select_query)
+		return FALSE
+
+	while(select_query.NextRow())
+		// Prep it all
+		var/banned_ckey = select_query.item[1]
+		var/banned_species = select_query.item[2]
+
+		if(!GLOB.species_bans_assoc["[banned_ckey]"]) // Incase someone has a ckey thats just all numbers
+			GLOB.species_bans_assoc["[banned_ckey]"] = list()
+
+		GLOB.species_bans_assoc["[banned_ckey]"] |= banned_species // Incase someone gets banned twice
+
+	qdel(select_query)
+
+// Global so it can be used in other places (IE: Conversion to another species)
+/proc/is_species_banned(ckey, species)
+	ASSERT(ckey)
+	ASSERT(species)
+	if(!GLOB.species_bans_assoc["[ckey]"]) // If they dont have an entry in the glob, they aint banned
+		return FALSE
+
+	// Cast to local list
+	var/list/species_cache = GLOB.species_bans_assoc["[ckey]"]
+	if(species in species_cache)
+		// If they in the list, they banned from it
+		return TRUE
+
+	// If we here, they aint banned
+	return FALSE
+
+
+/mob/verb/display_species_bans()
```

Comment:
Please no more verbs.
If you want to give players a way to check their species bans, build it into the existing check job bans verb. Please don't add a new verb and force the player to use both.

Reply:
Jobs and species are not the same thing, and I figured actual separation would be a good idea, especially since the `OOC` tab isn't used much at all. Its not the same problem as the `Admin` tab

Reply:
There should just be a single "Check Bans" verb that displays both. There are already too many verbs.